src/mscorlib/shared/System/Guid.cs

```diff
@@ -548,7 +548,7 @@ private static bool TryParseGuidWithHexPrefix(ReadOnlySpan<char> guidString, ref
 
             // Find the end of this hex number (since it is not fixed length)
             numStart = 3;
-            numLen = guidString.IndexOf(',', numStart) - numStart;
+            numLen = guidString.Slice(numStart).IndexOf(',') - numStart;
```

Comment:
Have you runs our tests, and does this pass?  Putting in the slice will change the resulting value as compared to passing in a start index.

Reply:
Same comment for all of these.

Reply:
I ran the corefx System.Runtime tests locally. I didn't run the coreclr tests.

I will fix the bug and run the coreclr tests.

Reply:
I actually meant the corefx tests.  If we don't have a test that catches this, we should.  This is wrong, right?  Consider if numStart is 3 and the ',' is found at 4.  Previously `IndexOf(',',numStart)` would return 4, and then we'd subtract 3 to get 1.  Now IndexOf(',') will return 1, and then we'll subtract 3 to get -2.

Reply:
Yes, it is incorrect. We need to exclude the `- numStart`

Reply:
We have [these](https://github.com/dotnet/corefx/blob/master/src/System.Runtime/tests/System/GuidTests.cs#L151) tests that, theoretically, should call into [TryParseGuidWithHexPrefix](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Guid.cs#L506). I will re-run the tests to see if they fail.

Reply:
Yep, the tests catch this bug and do fail. I must not have re-built coreclr successfully with my changes earlier (probably due to the issue with HW_Flag_BaseTypeFromFirstArg).